tools/state-viewer/src/cli.rs

```diff
@@ -606,3 +620,214 @@ impl ViewTrieCmd {
+        let iter = match iter {
+            Ok(iter) => iter,
+            Err(err) => {
+                println!("iter error {err:#?}");
+                return;
+            }
+        };
+        for item in iter {
+            node_count += 1;
+            if let Some(limit) = self.limit {
+                if limit < node_count {
+                    break;
+                }
+            }
+
+            let (key, value) = match item {
+                Ok((key, value)) => (key, value),
+                Err(err) => {
+                    println!("Failed to iterate node with error: {err}");
+                    error_count += 1;
+                    continue;
+                }
+            };
+
+            if self.print {
+                let state_record = StateRecord::from_raw_key_value(key.clone(), value);
+                Self::print_state_record(&state_record);
+            }
+        }
+        let duration = start.elapsed();
+        println!("node count  {node_count}");
+        println!("error count {error_count}");
+        println!("time        {duration:?}");
+    }
+
+    fn shallow_iter_prune_condition(key_nibbles: &Vec<u8>) -> bool {
+        // Need at least 2 nibbles for the column type byte.
+        if key_nibbles.len() < 2 {
+            return false;
+        }
+        // Keys for interesting nodes have even length
+        if key_nibbles.len() % 2 != 0 {
+            return false;
+        }
+
+        let col: u8 = key_nibbles[0] * 16 + key_nibbles[1];
+        match col {
+            // key for account only contains account id, nothing to prune
+            col::ACCOUNT => return false,
+            // key for contract code only contains account id, nothing to prune
+            col::CONTRACT_CODE => return false,
+            // key for delayed receipt only contains account id, nothing to prune
+            col::DELAYED_RECEIPT => return false,
+            // key for delayed receipt indices is a shard singleton, nothing to prune
+            col::DELAYED_RECEIPT_INDICES => return false,
+
+            // Most columns use the ACCOUNT_DATA_SEPARATOR to indicate the end
+            // of the accound id in the trie key. For those columns the
+            // partial_parse_account_id method should be used.
+            // The only exception is the ACCESS_KEY and dedicated method
+            // partial_parse_account_id_from_access_key should be used.
+            col::ACCESS_KEY => return Self::partial_parse_account_id_from_access_key(key_nibbles),
+            col::CONTRACT_DATA => return Self::partial_parse_account_id(col, key_nibbles),
+            col::RECEIVED_DATA => return Self::partial_parse_account_id(col, key_nibbles),
+            col::POSTPONED_RECEIPT_ID => return Self::partial_parse_account_id(col, key_nibbles),
+            col::PENDING_DATA_COUNT => return Self::partial_parse_account_id(col, key_nibbles),
+            col::POSTPONED_RECEIPT => return Self::partial_parse_account_id(col, key_nibbles),
```

Comment:
a suggestion is to add a feature to print out the counts of each of these kinds of nodes you've seen. That way we know we aren't missing any, and also have a rough idea what the scale of things are.